KuringApp/KuringApp/Search/SearchView.swift

```diff
@@ -33,9 +53,21 @@ struct SearchFeature: Reducer {
     enum Action: BindableAction {
         /// 최근 검색어 전체 삭제
         case deleteAllRecentsButtonTapped
+        /// 검색
+        case search
+        /// 검색된 공지
+        case fetchNotices([Notice])
+        /// 검색된 스태프
+        case fetchStaffs([Staff])
+        /// 검색 에러 핸들러
+        case searchFailure(State.SearchInfo.SearchType)
```

Comment:
```suggestion
        case searchCompletion(SearchResult)
        /// 검색 에러 핸들러
        case searchFailure(State.SearchInfo.SearchType)
```
```swift
enum SearchResult {
    case notices([Notice])
    case staffs([Staff])
}
```
이렇게 associated value에 enum을 넣으면 어떨까요.

그리고 이건 호불호가 있겠지만 Result 타입으로 한번더 감싸는 추가적인 방법도 있습니다
```swift
case searchResponse(Result<SearchResult, State.SearchInfo.SearchType>)

// 또는

case searchResponse(TaskResult<SearchResult>)
```

Reply:
https://github.com/ku-ring/ios-app/commit/df1b4b7f3377cbec0f8b678280c181fe5dea0466

에서 수정